components/nimbus/src/enrollment.rs

```diff
@@ -634,13 +638,16 @@ impl<'a> EnrollmentsEvolver<'a> {
         Ok(enrollments_change_events)
     }
 
-    pub(crate) fn evolve_enrollments(
+    pub(crate) fn evolve_enrollments<E>(
         &self,
         is_user_participating: bool,
-        prev_experiments: &[Experiment],
+        prev_experiments: &[E],
         next_experiments: &[Experiment],
```

Comment:
do you think this type should also be changed to `E`

Reply:
Not _yet_, but it's something we could do in the future. At the moment, the intent was only to add support for the `EnrolledFeatureConfig` type on `prev_experiments`. This plan is actually flawed as pointed out by @jhugman, but I'm going to address that in a [future ticket](https://mozilla-hub.atlassian.net/browse/EXP-3430).